File: Java/src/test/java/org/samples/Line.java
package org.samples;

import java.util.ArrayList;
import java.util.List;

public class Line {
    private List<Part> parts;

    public Line() {
        this.parts = new ArrayList<>();
    }

    public static Line keep(String text) {
        Line line = new Line();
        line.parts.add(new Part(text, "unchanged"));
        return line;
    }

    public Line remove(String text) {
        this.parts.add(new Part(text, "removed"));
        return this;
    }

    public Line add(String text) {
        this.parts.add(new Part(text, "added"));
        return this;
    }

    public Line keepPart(String text) {
        this.parts.add(new Part(text, "unchanged"));
        return this;
    }

    public List<Part> getParts() {
        return parts;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < parts.size(); i++) {
            sb.append(parts.get(i).toString());
            if (i < parts.size() - 1) {
                sb.append(", ");
            }
        }
        return sb.toString();
    }
}

File: Python/project/line.py
from project.part import Part


class Line:



    def __init__(self):
        self.parts = []

    def keep(self, text):
        self.parts.append(Part(text, 'unchanged'))
        return self

    def remove(self, text):
        self.parts.append(Part(text, 'removed'))
        return self

    def add(self, text):
        self.parts.append(Part(text, 'added'))
        return self

    def __str__(self):
        return ', '.join([str(part) for part in self.parts])